revoke/revoke.go

```diff
@@ -123,48 +263,90 @@ func getIssuer(cert *x509.Certificate) *x509.Certificate {
 	}
 
 	return issuer
-
 }
 
-// check a cert against a specific CRL. Returns the same bool pair
-// as revCheck.
-func certIsRevokedCRL(cert *x509.Certificate, url string) (revoked, ok bool) {
-	crl, ok := CRLSet[url]
+// checks whether CRL in memory is valid
+func (r *Revoke) isInMemoryCRLValid(key string) bool {
+	crl, ok := r.crl.set[key]
 	if ok && crl == nil {
 		ok = false
-		delete(CRLSet, url)
+		delete(r.crl.set, key)
+	} else if crl == nil {
```

Comment:
this 'if else' block is very confusing.

if ok == false and crl == nil, it means hashmap miss, and we do delete?